### ISSUE TITLE
How to set the minimum number of instances in one node

### ISSUE THREAD

**Issue by LNA28** [dmlc/xgboost#6191]
Hi, 

for data protection reasons I want to make sure that splitting stops when a minimum number of observation in one node is reached. In other tree boosting implementations there is a parameter to specify this. In case of Xgboost the manual says that min_child_weight specifys the minimum number ob observations in regression mode. 

Is there a possibility to specify the minimum number of observations in classification mode? 

Thank you for your help.

**Comment by hcho3** [comment]
`min_child_weight` is a useful proxy for the minimum number of observations per node. Currently, you cannot set the minimum number of observations per node directly.

**Comment by LNA28** [comment]
Thank you for your answer. Can you explain what exactly do you mean by proxy? If I set min_child_weight to lets say 10, is it possible that I end up with one single observation in one node with weight > 10? Or are all weights below 1, so maybe the condition min_child_weight=10 will be met when far more than 10 observations are in one node?

**Comment by hcho3** [comment]
The sum of Hessian weights is not necessarily equal to the number of instances. This is what I meant.

**Comment by LNA28** [comment]
But is the first scenario I described 

> If I set min_child_weight to lets say 10, is it possible that I end up with one single observation in one node with weight > 10
 also possible? How large can one weight be? I need to make sure nothing like this happens for non-disclosure reasons.

**Comment by hcho3** [comment]
This is also possible. You should look at the specific loss function you are using and see if the Hessian can exceed 10.

**Comment by hcho3** [comment]
FYI, if you are using binary:logistic, the Hessian weight value is given by

p * (1 - p)

where p is the probability prediction outputed by the tree ensemble.
